File: codegen/results/v002/qwen3.5_4b-nvfp4_nothink/20260412_083216/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
     """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
     """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75

def main():
    # Define expected values based on the logic
    test_cases = [
        {"price": 10.0, "quantity": 9, "expected": 90.0, "desc": "no discount (boundary < 10)"},
        {"price": 10.0, "quantity": 10, "expected": 90.0, "desc": "start of 10% discount"},
        {"price": 10.0, "quantity": 49, "expected": 441.0, "desc": "end of 10% discount"},
        {"price": 10.0, "quantity": 50, "expected": 375.0, "desc": "start of 25% discount"}
    ]

    print("Running tests for calculate_discount...")

    for test in test_cases:
        expected = test["expected"]
        desc = test["desc"]
        qty = test["quantity"]
        price = test["price"]
        
        actual = calculate_discount(price, qty)
        
        # Using a small epsilon for float comparison safety, though inputs are clean here
        if abs(actual - expected) < 1e-9:
            print(f"PASS: {desc}")
        else:
            print(f"FAIL: {desc} (expected: {expected}, got: {actual})")

if __name__ == "__main__":
    main()

File: codegen/results/v002/qwen3.5_4b-nvfp4_nothink/20260412_083216/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Test cases covering boundary conditions:
// 1. quantity = 9 (should use 0.0 multiplier)
// 2. quantity = 10 (should use 0.9 multiplier)
// 3. quantity = 49 (should use 0.9 multiplier)
// 4. quantity = 50 (should use 0.75 multiplier)

const testCases = [
    { name: 'quantity = 9', price: 10, expected: 90 },
    { name: 'quantity = 10', price: 10, expected: 90 },
    { name: 'quantity = 49', price: 10, expected: 441 },
    { name: 'quantity = 50', price: 10, expected: 375 },
];

let passedCount = 0;
let failedCount = 0;

for (const testCase of testCases) {
    const actualResult = calculateDiscount(testCase.price, testCase.quantity);
    const description = `price=${testCase.price}, quantity=${testCase.quantity}`;
    
    if (actualResult === testCase.expected) {
        console.log(`PASS: ${description}`);
        passedCount++;
    } else {
        console.log(`FAIL: ${description} (expected: ${testCase.expected}, got: ${actualResult})`);
        failedCount++;
    }
}

console.log(`\nResults: ${passedCount} passed, ${failedCount} failed.`);